dds/DCPS/security/CryptoBuiltInImpl.cpp

```diff
@@ -1774,36 +1958,41 @@ bool CryptoBuiltInImpl::decode_serialized_payload(
   mb_in.wr_ptr(encoded_buffer.length());
   Serializer de_ser(&mb_in, Serializer::SWAP_BE, Serializer::ALIGN_CDR);
   CryptoHeader ch;
-  de_ser >> ch;
+  if (!(de_ser >> ch)) {
+    return CommonUtilities::set_security_error(ex, -3, 4, "Failed to deserialize CryptoHeader");
+  }
 
   const KeySeq& keyseq = keys_[sending_datawriter_crypto];
```

Comment:
See question above. Should we check to see if the key exists in key_ here before accidentally adding a default one?